src/gui/plugins/lights/Lights.cc

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "Lights.hh"
+
+#include <ignition/msgs/boolean.pb.h>
+#include <ignition/msgs/stringmsg.pb.h>
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+
+#include <ignition/common/Console.hh>
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/MainWindow.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/transport/Node.hh>
+#include <ignition/transport/Publisher.hh>
+
+#include "ignition/gazebo/EntityComponentManager.hh"
+#include "ignition/gazebo/gui/GuiEvents.hh"
+
+namespace ignition::gazebo
+{
+  class LightsPrivate
+  {
+    /// \brief Ignition communication node.
+    public: transport::Node node;
+
+    /// \brief Mutex to protect mode
+    public: std::mutex mutex;
+
+    /// \brief Transform control service name
+    public: std::string service;
+  };
+}
+
+using namespace ignition;
+using namespace gazebo;
+
+/////////////////////////////////////////////////
+Lights::Lights()
+  : ignition::gui::Plugin(),
+  dataPtr(std::make_unique<LightsPrivate>())
+{
+}
+
+/////////////////////////////////////////////////
+Lights::~Lights() = default;
+
+/////////////////////////////////////////////////
+void Lights::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Lights";
+
+  // For shapes requests
+  ignition::gui::App()->findChild
+    <ignition::gui::MainWindow *>()->installEventFilter(this);
+}
+
+/////////////////////////////////////////////////
+void Lights::OnMode(const QString &_mode)
+{
+  std::string modelSdfString = _mode.toStdString();
```

Comment:
It's weird to prefix all of these variables with `model`, because there are no models here, these are lights. I think this could just be `sdfString` to save confusion.

I know that the event has `Model` in the name, but that's changing to be more generic in #595